src/framework/shortcuts/view/shortcutsmodel.cpp

```diff
@@ -116,7 +134,9 @@ bool ShortcutsModel::apply()
     ShortcutList shortcuts;
 
     for (const Shortcut& shortcut : qAsConst(m_shortcuts)) {
-        shortcuts.push_back(shortcut);
+        if (!shortcut.sequences.empty()) {
```

Comment:
@VanSHOE 
Can you explain to me why you filter empty shortcuts here?

Reply:
Otherwise we fill the shortcuts register (and thus the user shortcuts xml file) with _all_ actions that exist, while only a few have a shortcut.

However, @VanSHOE and me are aware that this solution is not perfect, because if the user deletes a default shortcut, this deletion is not saved.

So maybe we should accept that we store a big amount of unnecessary information in the shortcuts register. 

Or we should make this filter clever, by only saving user shortcuts that differ from the default ones instead of only saving non-empty user shortcuts. Maybe that's the best solution.

Reply:
There is another issue which occurs even if we do store the ones that are currently different from the default ones. 
If a default shortcut is added in the development process to the xml which was previously not assigned anything, it would not get assigned to any one if we choose store empty values in the register.

I was thinking that we have another flag with each shortcut where we can mark if it was user cleared. In that case, we do write it to the shortcuts register.

Other shortcuts that are just not assigned any shortcut can continue to not exist in the register.

Reply:
There are more problems than benefits. I will revert this changes in https://github.com/musescore/MuseScore/pull/12890